architecture/azure/messaging.md

```diff
@@ -0,0 +1,97 @@
+
+### Azure Queue Storage
+
+[Azure Queue Storage](https://learn.microsoft.com/en-us/azure/storage/queues/storage-queues-introduction) offer a simple, low-cost, and easy-to-use messaging service based on the Azure Storage infrastructure.
+
+:heavy_plus_sign: Pros:
+
+- Queues can contain up to 200 TB of messages
+- Cheaper than Azure Service Bus
+- Part of Azure Storage
+- The Azure Storage Emulator may be on-premises for development, testing, and CI
+
+:heavy_minus_sign: Cons:
+
+- The maximum message size is 64 KB
+- Higher latency due to HTTP communication
+- No cross-entity transactions
+- The maximum number of requests (e.g. message delivery) per storage account is 20,000 per second (2,000 per second for single queues)
+
+[**Try the Azure Storage Queues sample →**](/samples/azure/storage-queues/)
+
+#### When to use Azure Storage Queues
+
+Consider Azure Storage Queues when a managed message queuing technology is required but Azure Service Bus doesn't meet requirements or is too expensive.
+
+### SQL transport
+
+SQL transport is an NServiceBus feature that can use existing SQL Server-compatible databases as feature-complete message queues.
+
+:heavy_plus_sign: Pros:
+
+- Runs on cloud hosted and on-premises SQL Server-compatible data stores (including SQL Server Express for development, testing, and CI)
+- Arbitrary message sizes
+- Runs on infrastructure which often already exists
+- Strong transaction integration with business data operations
+
+:heavy_minus_sign: Cons:
+
+- More expensive and laborious to scale
+- Impacts overall database performance
+- Lower message throughput compared to specialized message queuing technologies
+
+[**Try the SQL transport sample →**](/samples/azure/storage-queues/)
+
+#### When to use SQL transport
+
+Consider SQL transport if an existing application already uses a SQL Server-compatible data store and limited amount of messaging is being introduced. SQL transport can be a good stepping-stone when introducing messaging into an existing system without the introduction of new infrastructure.
+
+## Other services
+
+Azure offers other messaging services focused on asynchronous communication. These services are not directly support by the Particular Platform, however they might be combined in combination with supported message queueing systems to handle [data distribution](/nservicebus/concepts/data-distribution.md) needs by more specialized technologies.
+
+* [Azure Event Grid](https://learn.microsoft.com/en-us/azure/event-grid/overview) is an integration focused messaging system using the publish-subscribe model. Azure Event Grid can ingest and distribute events provided by applications, other Azure Services or IoT devices using HTTP protocols (MQTT support in public preview). By default, Event Grid uses a [push-model](https://learn.microsoft.com/en-us/azure/event-grid/push-delivery-overview) to forward events to all subscribers.
```

Comment:
With Event Grid v2 this needs to change IMO.

Ideally, NSB should have a transport EG with [pull-based delivery](https://learn.microsoft.com/en-us/azure/event-grid/pull-delivery-overview). It's not as polished and rounded as ASB but it will be there. And for customers that are more cloud-native/first, this transport would be a natural option.

Reply:
I'll make an internal note to keep this in mind and update once the new APIs move out of preview